prereise/gather/demanddata/transportation_electrification/generate_BEV_vehicle_profiles.py

```diff
@@ -0,0 +1,146 @@
+    model_year,
+    state,
+    load_demand,
+    power=6.6,
+    location_strategy=2,
+    trip_strategy=1,
+):
+    """Generate Battery Electric Vehicle (BEV) profiles
+
+    :param str vehicle_trip_data_filepath: collected trip data from external sources representing driving patterns
+    :param str charging_strategy: establishes whether charging happens immediately or via a smart charging algorithm;
+                current options are "immediate" and "smart"
+    :param str veh_type: vehicle category (LDV, LDT, MDV, HDV)
+    :param int veh_range: 100, 200, or 300, represents how far vehicle can travel on
+        single charge.
+    :param int model_year: year that is being modelled/projected to, 2017, 2030, 2040,
+        2050.
+    :param str state: US state abbreviation
+    :param np.array load_demand: initial load demand (MW for each hour)
+    :param int power: (optional) charger power, EVSE kW; default value: 6.6 kW;
+    :param int location_strategy: (optional) where the vehicle can charge-1, 2, 3, 4, or 5;
+        1-home only, 2-home and work related, 3-anywhere if possibile,
+        4-home and school only, 5-home and work and school.
+        default value: 2
+    :param int trip_strategy: (optional) determine to charge after any trip (1) or only after the
+        last trip (2); default value: 1
+    :return: (*numpy.ndarray*) -- yearly charging profiles for all urban areas and the rural area in each state (MW for each hour)
+    """
+
+    census_region = const.state2census_region[state]
+    kwhmi = get_kwhmi(model_year, veh_type, veh_range)
+
+    daily_values = generate_daily_weighting(model_year)
+
+    if power > 19.2:
+        charging_efficiency = 0.95
+    else:
+        charging_efficiency = 0.9
+
+    geographic_area_bev_vmt = {}
+
+    urban_scaling_filepath = os.path.join(
+        const.data_folder_path,
+        "regional_scaling_factors",
+        "Regional_scaling_factors_UA_",
+    )
+    urban_scaling_factors = pd.read_csv(
+        urban_scaling_filepath + str(model_year) + ".csv", index_col=["State"]
+    )
+    state_urban_areas = urban_scaling_factors.loc[state, "UA"]
+
+    # scaling factors for listed urban areas
+    for urban_area in state_urban_areas:
+        urban_bev_vmt = load_urbanized_scaling_factor(
+            model_year=model_year,
+            veh_type=veh_type,
+            veh_range=veh_range,
+            urbanized_area=urban_area,
+            state=state,
+            filepath=urban_scaling_filepath,
+        )
+        geographic_area_bev_vmt.update({urban_area: urban_bev_vmt})
+
+    # scaling factors for rural areas
+    rural_bev_vmt = load_rural_scaling_factor(
+        model_year,
+        veh_type,
+        veh_range,
+        state,
+        filepath=os.path.join(
+            const.data_folder_path,
+            "regional_scaling_factors",
+            "Regional_scaling_factors_RA_",
+        ),
+    )
+    geographic_area_bev_vmt.update({state: rural_bev_vmt})
```

Comment:
Should we add a suffix to this key to specify "the rural area of the state", say `f"{state}_rural"`?

Reply:
Given the discussion about urban areas being in more than one state, I think we should consider adding the `state` name to all of these.
E.g.
`f"{state}_rural"`
`f"{state}_urban_area"'